pyrefly/lib/alt/solve.rs

```diff
@@ -336,6 +336,31 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
                         members
                     ),
                 );
+            } else if !metadata.is_protocol() {
+                let has_direct_abc_base = metadata
+                    .base_class_objects()
```

Comment:
I belive you can just check `extends_abc` on the class metadata do do the same thing as has_direct_abc_base + has_explicit_abc_metaclass

https://github.com/facebook/pyrefly/blob/ed6224fd4783eabaeb8e07ee063ffa8c96f2d1e5/pyrefly/lib/alt/class/class_metadata.rs#L1158

I'd expect the logic change here to be fairly minimal

basically just

```
if is_final:
   old error
else if !is_protocol && !extends_abc
   new error
```